app/Http/Controllers/Api/AssetsController.php

```diff
@@ -798,4 +815,42 @@ public function requestable(Request $request)
         $assets = $assets->skip($offset)->take($limit)->get();
         return (new AssetsTransformer)->transformRequestedAssets($assets, $total);
     }
+
+    /**
+     * @param string $image_data base64 encoded image data with mime type
+     * @return string path to uploaded image
+     * @throws  \Exception  if image source is not in correct format
+     */
+    private function image(String $image_data)
```

Comment:
Sorry for the delay in finishing this review. We should also probably rename this method, since it actually performs an action, and return false if something goes wrong. Something like processUploadedImage, or something that's a bit more specific as to what it's doing. In this massive codebase, I'm not sure I'll ever remember that `->image()` actually processes things, and could be easily confused with the image property of many of our models.

Reply:
Also should this maybe be a helper, trait or model method? Lots of our models have images. If there's a way to break it out so that it could be re-used, that would probably be more useful.

Reply:
The method now returns false if something goes wrong. So it's much more general while the endpoint still retains the possibility to wipe the image from hardware if `image_source` field is set but empty.